extras/nestml-release-checklist.md

```diff
@@ -0,0 +1,41 @@
+Release checklist
+-----------------
+
+Follow this checklist to successfully perform a NESTML release. Let's say that 3.0 is the currently-out release, and we want to publish 3.1.
+
+- Find out authors who contributed since the last release.
+
+  ```bash
+  git log v3.0..HEAD | grep Author\: | sort | uniq
+  ```
+
+- Ensure copyright transferal agreement is obtained from each author (contact: HEP).
+
+- Edit `setup.py` and enter the right version number.
+
+- Add a tag to the repository. Since v3.0 we're adding the "v" prefix.
```

Comment:
I suggest to create the release on GitHub instead. That will do the tagging for you and also provides a tarball and zipfile.